nuxeo-core/nuxeo-core-api/src/test/java/org/nuxeo/ecm/core/api/impl/TestDocumentModelWithSecuredProperty.java

```diff
@@ -0,0 +1,203 @@
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.RuntimeFeature;
+
+/**
+ * @since 9.1
+ */
+@RunWith(FeaturesRunner.class)
+@Features({ RuntimeFeature.class, DummyLoginFeature.class })
+@Deploy("org.nuxeo.ecm.core.schema")
+@Deploy("org.nuxeo.ecm.core.api.tests:OSGI-INF/test-documentmodel-secured-types-contrib.xml")
+@WithUser("john")
+public class TestDocumentModelWithSecuredProperty {
+
+    protected static final String SECURED_SCHEMA = "secured";
+
+    @Test
+    public void testSetSecuredScalarProperty() {
+        try {
+            DocumentModel doc = new DocumentModelImpl("/", "doc", "Secured");
+            doc.setProperty(SECURED_SCHEMA, "scalar", "test secure");
+            fail("A ReadOnlyPropertyException should have been thrown");
+        } catch (ReadOnlyPropertyException e) {
+            assertEquals("scalar", e.getMessage());
+        }
+    }
+
+    @Test
+    public void testSetSecuredScalarPropertyWithSystem() {
+        Framework.doPrivileged(() -> {
+            DocumentModel doc = new DocumentModelImpl("/", "doc", "Secured");
+            doc.setProperty(SECURED_SCHEMA, "scalar", "test secure");
+            assertEquals("test secure", doc.getProperty(SECURED_SCHEMA, "scalar"));
+        });
+    }
+
+    @Test
+    @WithUser("Administrator")
+    public void testSetSecuredScalarPropertyWithAdministrator() {
+        DocumentModel doc = new DocumentModelImpl("/", "doc", "Secured");
+        doc.setProperty(SECURED_SCHEMA, "scalar", "test secure");
+        assertEquals("test secure", doc.getProperty(SECURED_SCHEMA, "scalar"));
+    }
+
+    @Test
+    public void testSetUnsecuredScalarProperty() {
+        DocumentModel doc = new DocumentModelImpl("/", "doc", "Secured");
+        doc.setProperty(SECURED_SCHEMA, "unsecureScalar", "test unsecure");
+        assertEquals("test unsecure", doc.getProperty(SECURED_SCHEMA, "unsecureScalar"));
+    }
+
+    @Test
+    public void testSetSecuredComplexProperty() {
+        try {
+            DocumentModel doc = new DocumentModelImpl("/", "doc", "Secured");
+            doc.setProperty(SECURED_SCHEMA, "complex", Map.of("scalar1", "test secure1", "scalar2", "test secure2"));
+            fail("A ReadOnlyPropertyException should have been thrown");
+        } catch (ReadOnlyPropertyException e) {
+            assertEquals("complex", e.getMessage());
+        }
+    }
+
+    @Test
+    @WithUser("Administrator")
+    public void testSetSecuredComplexPropertyWithAdministrator() {
+        DocumentModel doc = new DocumentModelImpl("/", "doc", "Secured");
+        doc.setProperty(SECURED_SCHEMA, "complex", Map.of("scalar1", "test secure1", "scalar2", "test secure2"));
+        assertEquals("test secure1", doc.getProperty(SECURED_SCHEMA, "complex/scalar1"));
+        assertEquals("test secure2", doc.getProperty(SECURED_SCHEMA, "complex/scalar2"));
+    }
+
+    @Test
+    public void testSetUnsecuredComplexProperty() {
```

Comment:
`testSetUnsecuredComplexItemProperty`, as the test below?

Reply:
Probably not, I use `Item` to state that it's the item which is secured or not, in this test this is the complex property which is unsecure while in `testSetSecuredComplexItemProperty` this is the item inside the complex which is secure will review the whole test class.